text/0036-ternary.md

```diff
@@ -0,0 +1,280 @@
+
+Ternary has been a fundamental part of the IOTA technology and is used
+throughout, including for fundamental features like wallet addresses and transaction
+identification.
+
+More information about ternary in IOTA can be found here:
+(https://docs.iota.org/docs/getting-started/0.1/introduction/ternary)
+
+*Note that the IOTA foundation is seeking to reduce the prevalence of ternary
+in the IOTA protocol to better accomodate modern binary architectures. See
+[Chrysalis](https://blog.iota.org/chrysalis-b9906ec9d2de) for more information.*
+
+Manipulating ternary in an ergonomic manner on top of existing binary-driven
+code can require complexity and, up until now, comes either with a lot of
+boilerplate code, implementations that are difficult to verify as correct, or a
+lack of features.
+
+`bee-ternary` seeks to become the canonical ternary crate in the Rust IOTA
+ecosystem by providing an API that is efficient, ergonomic, and featureful all
+at once.
+
+We hope that `bee-ternary` will allow IOTA developers to more easily write code
+that works with ternary and allow the simplification of existing codebases
+through use of the API.
+
+Broadly, there are 3 main benefits that `bee-ternary` aims to provide
+over use-specific implementations of ternary manipulation code.
+
+- **Ergonomics**: The API should be trivial to use correctly and should
+naturally guide developers towards correct and efficient solutions.
+
+- **Features**: The API should provide a fundamental set of core features that
+can be used together to cover most developer requirements. Examples of such
+features include multiple encoding schemes, binary/ternary conversion,
+(de)serialization, and tryte string de/encoding.
+
+- **Performance**: The API should allow 'acceptably' efficient manipulation of
+ternary data. This is difficult to ruggedly define, and implementation details
+may change later, but broadly we intend to introduce an API that does not
+inherently inhibit high performance through poor design choices.
+
+# Detailed design
+
+`bee-ternary` is designed to be extensible and is built on top of a handful of
+fundamental abstraction types. Below we list the features of the API and
+summarize a few of the core API features.
+
+## Features
+
+- Efficient manipulation of ternary buffers (trits and trytes)
+- Multiple encoding schemes
+- Extensible design that allows it to sit on top of existing data structures,
+avoiding unnecessary allocation and copying
+- An array of utility functions to allow for easy manipulation of ternary data
+- Zero-cost conversion between trit and tryte formats
+
+## Key Types & Traits
+
+### `Btrit` and `Utrit`
+
+```rust
+enum Btrit { NegOne, Zero, PlusOne }
+enum Utrit { Zero, One, Two }
+```
+
+The crate supports both balanced and unbalanced trit representations, supported
+through two types, `Btrit` and `Utrit`. `Btrit` is the more common
+representation and so is the implicit default for most operations in the crate.
+
+### `Trits`
+
```

Comment:
Maybe it's just me but I wished there were a short introductory sentence for each sub-section before showing code. I do not wish to bloat the RFC. I like a minimalistic approach, but still....a short paragraph explaining the reasoning would help with the understanding why this RFCs suggests a certain abstraction.

Reply:
I'm not sure what information I could provide here. Do you have any specific ideas?